src/utils.js

```diff
@@ -307,392 +307,47 @@ _.strip_empty_properties = function (p) {
     return ret
 }
 
-/*
- * this function returns a copy of object after truncating it.  If
- * passed an Array or Object it will iterate through obj and
- * truncate all the values recursively.
- */
-_.truncate = function (obj, length) {
-    var ret
-
-    if (typeof obj === 'string') {
-        ret = obj.slice(0, length)
-    } else if (_.isArray(obj)) {
-        ret = []
-        _.each(obj, function (val) {
-            ret.push(_.truncate(val, length))
-        })
-    } else if (_.isObject(obj)) {
-        ret = {}
-        _.each(obj, function (val, key) {
-            ret[key] = _.truncate(val, length)
+// Deep copies an object.
+// It handles cycles by replacing all references to them with `undefined`
+// Also supports customizing native values
+function deepCircularCopy(value, customizer) {
+    const gdcc = '__deepCircularCopyInProgress__'
```

Comment:
This seems like a good place to use [Symbols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) to avoid any (obviously really rare) key name conflicts.

However... yep. IE :-)

Some `(typeof Symbol === 'undefined' ? String : Symbol)` style polyfill could work, though adding this might go down a rabbithole fast.

Up to you if worth it.

Reply:
Good idea. Added a solution here but didn't really test it out on an older browsers.